components/match2/commons/match_group_legacy.lua

```diff
@@ -6,259 +6,364 @@
 -- Please see https://github.com/Liquipedia/Lua-Modules to contribute
 --
 
-local Legacy = {}
-
 local Arguments = require('Module:Arguments')
+local Array = require('Module:Array')
+local Class = require('Module:Class')
 local Json = require('Module:Json')
 local Logic = require('Module:Logic')
 local Lua = require('Module:Lua')
-local MatchGroup = require('Module:MatchGroup')
+local PageVariableNamespace = require('Module:PageVariableNamespace')
 local String = require('Module:StringUtils')
 local Table = require('Module:Table')
-local Variables = require('Module:Variables')
 
-local MatchGroupLegacyDefault = Lua.import('Module:MatchGroup/Legacy/Default')
+local MatchGroup = Lua.import('Module:MatchGroup')
 local MatchSubobjects = Lua.import('Module:Match/Subobjects')
-local WikiSpecific = Lua.import('Module:Brkts/WikiSpecific')
 
-local IS_USERSPACE = false
-local NAMESPACE_USER = 2
+local globalVars = PageVariableNamespace()
+
+local MAX_NUMBER_OF_OPPONENTS = 2
 local RESET_MATCH = 'RxMBR'
 local THIRD_PLACE_MATCH = 'RxMTP'
-local _args
-local _type
 
----@param frame Frame
+---@alias roundKeys {R: number, G: number, W: number, D: number}
+---@alias match1Keys {opp1: string, opp2: string, details: string}
+---@alias match2mapping {[string]: match1Keys}|{[string] : string}
+
+---@class MatchGroupLegacy
+---@operator call(table): MatchGroupLegacy
+---@field args table
+---@field bracketType string
+---@field newArgs table
+---@field mapMappings {[number]: table}
+local MatchGroupLegacy = Class.new(function(self, frame)
+	local args = Arguments.getArgs(frame)
+	assert(String.isNotEmpty(args.id), 'Argument \'id\' is empty')
+	assert(String.isNotEmpty(args.template), 'Argument \'template\' is empty')
+	assert(String.isNotEmpty(args.templateOld), 'Argument \'templateOld\' is empty')
```

Comment:
imo do not make this a necessity
like if the template follows the default mapping the old template name is irrelevant

Reply:
maybe move templateOld assertion to MatchGroupLegacy._get

Reply:
Tbh, after some tought I think we should keep it as required, as peace of mind.

Reply:
it will cause several errors on several wikis

Reply:
imo move the assert to when a custom exists
like if the module for the custom mapping is found assert the old template is set
that should mitigate most issues

Reply:
I'm fine with keeping tempateOld as requirement, as it is righ now. Does make it faster in case of debugging, or after post-live realization that a cerntain templateOld was converted incorrectly (eg due to user error in bot input)